docs/classifier_example/classification_v4_4_features.ipynb

```diff
@@ -0,0 +1,884 @@
+   "outputs": [],
+   "source": [
+    "a00, a01, a10, a11 = first_data_point\n",
+    "alpha1 = atan(sqrt((a10**2  + a11**2) / (a00**2  + a01**2)))\n",
+    "alpha2 = np.arctan2(a01, a00) \n",
+    "alpha3 = np.arctan2(a11, a10) "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As always the first step is to set up a connection with the Quantum Inspire:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Enter email:\n",
+      "demo@quantum-inspire.com\n",
+      "Enter password\n",
+      "········\n"
+     ]
+    }
+   ],
+   "source": [
+    "from getpass import getpass\n",
+    "from coreapi.auth import BasicAuthentication\n",
+    "\n",
+    "from qiskit.validation.base import Obj\n",
+    "from qiskit.circuit import QuantumRegister, ClassicalRegister, QuantumCircuit\n",
+    "from qiskit.tools.visualization import plot_state, plot_histogram, circuit_drawer\n",
+    "from qiskit import execute\n",
+    "\n",
+    "from quantuminspire.qiskit import QI\n",
+    "from quantuminspire.api import QuantumInspireAPI\n",
+    "\n",
+    "\n",
+    "def get_authentication():\n",
+    "    \"\"\" Gets the authentication for connecting to the Quantum Inspire API.\"\"\"\n",
+    "    if 'password' not in vars().keys():\n",
+    "        print('Enter email:')\n",
+    "        email = input()\n",
+    "        print('Enter password')\n",
+    "        password = getpass()\n",
+    "    return BasicAuthentication(email, password)\n",
+    "\n",
+    "authentication = get_authentication()\n",
+    "server_url = r'https://api.quantum-inspire.com'\n",
+    "\n",
+    "# Temporairy alternative so that we can give our project a name:\n",
+    "QI._api = QuantumInspireAPI(server_url, authentication, project_name=\"Distance-based Classifier more features (NEW)\")\n",
+    "\n",
+    "# Alternative:\n",
+    "# authentication = [email, password]\n",
+    "# QI.set_authentication_details(*authentication)\n",
+    "\n",
+    "qi_backend = QI.get_backend('QX single-node simulator')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can now construct a function which builds the quantum circuit as discussed above. Note that Qiskit contains the general unitary gates $u3(\\theta, \\phi, \\lambda)$ which are related to our definition of $U(\\alpha)$ by:$$U(\\alpha) = u3(2\\alpha, 0, \\pi).$$\n",
+    "Unfortunately, these $u3$ gates are not yet implemented on the quantum inspire backend and thus we need to make use of regular $R_y$ rotations and CNOT gates."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def cRy(q, angle, ctrlidx, idx):\n",
+    "    if len(q)<2:\n",
+    "        raise IndexError(\"Error, len quantum register must at least be 2.\")\n",
```

Comment:
is this not a value error?

Reply:
You're right, I'll change it to ValueError